core/src/main/scala/org/apache/spark/util/ListenerBus.scala

```diff
@@ -80,6 +89,11 @@ private[spark] trait ListenerBus[L <: AnyRef, E] extends Logging {
       }
       try {
         doPostEvent(listener, event)
+        if (Thread.interrupted()) {
```

Comment:
This is ok right now since Spark code never explicitly interrupts these threads. If we ever need to do that, though, this might become a problem... but in that case I don't know how you'd handle this issue without just giving up and stopping everything.

But... is this correct? Your test just calls `Thread.interrupt()`, which just sets the interrupt flag. But what if the listener calls an operation that would actually turn that into an `InterrruptedException`? You're not catching that here anywhere.

```
scala> Thread.currentThread().interrupt()

scala> Thread.sleep(10)
java.lang.InterruptedException: sleep interrupted
  at java.lang.Thread.sleep(Native Method)
  ... 33 elided
```

Reply:
> This is ok right now since Spark code never explicitly interrupts these threads. If we ever need to do that, though, this might become a problem... but in that case I don't know how you'd handle this issue without just giving up and stopping everything.

If spark were to explicitly interrupt, then I think we'd also set some other flag indicating a reason, eg. `val requestedQueueStop: AtomicBoolean` so it shouldn't be hard to distinguish.

I've pushed an update to handle `InterruptedException` as well.